### PR TITLE
feat: refactor construct knowledge graph task

### PR DESCRIPTION
**Pull request by simon824**
refactor construct knowledge graph task

### CODE REVIEW

**Comment by simon824** [comment]
@imbajin @Zony7 @liuxiaocs7 @javeme please take a review when free, thanks!

File: hugegraph-llm/README.md [hunk 1]
```diff
@@ -19,8 +19,64 @@ graph systems and large language models.
...
+    builder
+    .import_schema(from_hugegraph="talent_graph").print_result()
+    .extract_triples(TEXT).print_result()
+    .disambiguate_word_sense().print_result()
+    .commit_to_hugegraph()
+    .run()
+)
+```
+
+2. **Import Schema**: The `import_schema` method is used to import a schema from a source. The source can be a HugeGraph instance,a user-defined schema or an extraction result. The method `print_result` can be chained to print the result.
```
Comment (liuxiaocs7) [inline hugegraph-llm/README.md:47]:
```suggestion
2. **Import Schema**: The `import_schema` method is used to import a schema from a source. The source can be a HugeGraph instance, a user-defined schema or an extraction result. The method `print_result` can be chained to print the result.
```

File: hugegraph-llm/examples/build_kg_test.py [hunk 2]
```diff
@@ -31,47 +32,29 @@
...
+    }
 
     (
-        builder.parse_text_to_data_with_schemas(TEXT, nodes_schemas, relationships_schemas)
-        .disambiguate_data_with_schemas()
-        .commit_data_to_kg()
+        builder
+        .import_schema(from_hugegraph="talent_graph").print_result()
+        # .import_schema(from_extraction="fefe").print_result().run()
+        # .import_schema(from_input=schema).print_result()
```
Comment (liuxiaocs7) [inline hugegraph-llm/examples/build_kg_test.py:55]: Why does the comment here not match what is described in the documentation?

```python
# Import schema from a HugeGraph instance
import_schema(from_hugegraph="talent_graph").print_result()
# Import schema from an extraction result
import_schema(from_extraction="xxx").print_result()
# Import schema from user-defined schema
import_schema(from_user_defined="xxx").print_result()
```

File: hugegraph-llm/src/hugegraph_llm/operators/common_op/check_schema.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+class CheckSchema:
+    def __init__(self, data):
+        self.result = None
+        self.data = data
+
+    def run(self, schema=None) -> Any:
+        data = self.data or schema
+        if not isinstance(data, dict):
+            raise ValueError("Input data is not a dictionary.")
```
Comment (liuxiaocs7) [inline hugegraph-llm/src/hugegraph_llm/operators/common_op/check_schema.py:30]:
```suggestion
            raise ValueError("Input schema is not a dictionary.")
```

Emphasizing `schema` might be better? Same below.

File: hugegraph-llm/src/hugegraph_llm/operators/hugegraph_op/commit_to_hugegraph.py [hunk 4]
```diff
@@ -220,9 +104,9 @@ def run(self, data: dict):
         ).secondary().ifNotExist().create()
 
         for item in data:
-            s = item[0]
-            p = item[1]
-            o = item[2]
+            s = item[0].strip()
+            p = item[1].strip()
+            o = item[2].strip()
```
Comment (liuxiaocs7) [inline hugegraph-llm/src/hugegraph_llm/operators/hugegraph_op/commit_to_hugegraph.py:109]:
```suggestion
            s, p, o = (element.strip() for element in item)
```

WDYT?

File: hugegraph-llm/src/hugegraph_llm/operators/hugegraph_op/schema_manager.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        for el in schema["edgelabels"]:
+            edge = {
+                "edge_label": el["name"],
+                "source_vertex_label": el["source_label"],
+                "target_vertex_label": el["target_label"],
+                "properties": el["properties"],
+            }
+            edges.append(edge)
+        if not vertices and not edges:
+            raise Exception(f"Can not get {self.graph_name}'s schema from HugeGraph!")
```
Comment (liuxiaocs7) [inline hugegraph-llm/src/hugegraph_llm/operators/hugegraph_op/schema_manager.py:53]: The schema will fail if there are no edges in the graph?

File: hugegraph-llm/src/hugegraph_llm/operators/llm_op/info_extract.py [hunk 6]
```diff
@@ -165,110 +65,62 @@ def split_string_to_fit_token_space(
...
 
 
-def get_spo_from_result(result):
-    res = []
-    for row in result:
-        row = row.replace("\\n", "").replace("\\", "")
-        pattern = r'\("(.*?)", "(.*?)", "(.*?)"\)'
-        res += re.findall(pattern, row)
-    return res
+def extract_by_regex(text, triples):
```
Comment (liuxiaocs7) [inline hugegraph-llm/src/hugegraph_llm/operators/llm_op/info_extract.py:68]:
```suggestion
def extract_triples_by_regex(text, triples):
```

`v-n-adv` phrase maybe better?

File: hugegraph-llm/src/hugegraph_llm/operators/llm_op/info_extract.py [hunk 7]
```diff
@@ -165,110 +65,62 @@ def split_string_to_fit_token_space(
...
+    text = text.replace("\\n", " ").replace("\\", " ").replace("\n", " ")
+    pattern = r"\((.*?), (.*?), (.*?)\)"
+    triples["triples"] += re.findall(pattern, text)
+
 
+def extract_by_regex_with_schema(schema, text, graph):
```
Comment (liuxiaocs7) [inline hugegraph-llm/src/hugegraph_llm/operators/llm_op/info_extract.py:74]: same above?

File: hugegraph-llm/src/tests/operators/llm_op/test_disambiguate_data.py [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+                        "edge_label": "roommate",
+                        "source_vertex_label": "person",
+                        "target_vertex_label": "person",
+                        "properties": [],
+                    }
+                ],
+            },
+        }
+        self.llm = LLMs().get_llm()
+        # self.llm =  None
```
Comment (liuxiaocs7) [inline hugegraph-llm/src/tests/operators/llm_op/test_disambiguate_data.py:78]: remove?

File: hugegraph-llm/src/tests/operators/llm_op/test_disambiguate_data.py [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+
+from hugegraph_llm.llms.init_llm import LLMs
+from hugegraph_llm.operators.llm_op.disambiguate_data import DisambiguateData
+
+
+class TestDisambiguateData(unittest.TestCase):
+    def setUp(self):
+        self.triples = {
+            "triples": [
+                (' "Alice "', ' "Age "', ' "25 "'),
```
Comment (liuxiaocs7) [inline hugegraph-llm/src/tests/operators/llm_op/test_disambiguate_data.py:28]: Why do we need extra quotes and spaces? Can't `('Alice', 'Age', '25')` represent a SPO?

File: hugegraph-llm/src/tests/operators/llm_op/test_info_extract.py [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+            "edges": [
+                {
+                    "edge_label": "roommate",
+                    "source_vertex_label": "person",
+                    "target_vertex_label": "person",
+                    "properties": [],
+                }
+            ],
+        }
+        # self.llm = LLMs().get_llm()
```
Comment (liuxiaocs7) [inline hugegraph-llm/src/tests/operators/llm_op/test_info_extract.py:43]: remove?

File: hugegraph-llm/README.md (hunk 1 above)
Comment (liuxiaocs7) [reply hugegraph-llm/README.md:47]: also expect blank line in next line

File: hugegraph-llm/src/hugegraph_llm/operators/hugegraph_op/schema_manager.py (hunk 5 above)
Comment (simon824) [reply hugegraph-llm/src/hugegraph_llm/operators/hugegraph_op/schema_manager.py:53]: no vertices and no edges at the same time

Comment (liuxiaocs7) [reply hugegraph-llm/src/hugegraph_llm/operators/hugegraph_op/schema_manager.py:53]: Sorry for the misreading, please ignore this comment 😄

File: hugegraph-llm/src/tests/operators/llm_op/test_disambiguate_data.py (hunk 9 above)
Comment (simon824) [reply hugegraph-llm/src/tests/operators/llm_op/test_disambiguate_data.py:28]: Of course, these triples are output from ernie-bot, maybe more like real scenes?

File: hugegraph-llm/src/tests/operators/llm_op/test_disambiguate_data.py [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+                    }
+                ],
+            },
+        }
+        self.llm = None
+        self.disambiguate_data = DisambiguateData(self.llm)
+
+    def test_run(self):
+        result = self.disambiguate_data.run(self.triples_with_schema)
+        self.assertEqual(result, self.triples_with_schema)
```
Comment (Zony7) [inline hugegraph-llm/src/tests/operators/llm_op/test_disambiguate_data.py:81]: AssertDictEqual maybe better? I think  assertDictEqual method can ensure that the contents of the dictionary objects are compared. 
As follows:
`self.assertDictEqual(result, self.triples_with_schema)`

**Review by javeme — COMMENTED** [review]
+1

**Review by liuxiaocs7 — APPROVED** [review]

**Pull request merged**